index.js

```diff
@@ -0,0 +1,47 @@
+import * as THREE from 'three';
+import metaversefile from 'metaversefile';
+
+const {useFrame, useLocalPlayer, useCleanup, useMaterials, usePhysics} = metaversefile;
+
+export default () => {
+
+  const physics = usePhysics();
+
+  const rootScene = new THREE.Object3D();
+
+  const dims = [32, 32, 32];
+  const shift = [0, 0, 0];
+  const scale = [1.0, 1.0, 1.0];
+  let potential = [];
+
+  const center = new THREE.Vector3(...dims).multiplyScalar(0.5);
+
+  for (let x = 0; x < dims[0]; x++) {
+    for (let y = 0; y < dims[1]; y++) {
+      for (let z = 0; z < dims[2]; z++) {
+        potential[x * dims[0] * dims[1] + y * dims[0] + z] = -1;
+        if (new THREE.Vector3(x, y, z).distanceTo(center) < dims[0] / 2 - 1) {
+          potential[x * dims[0] * dims[1] + y * dims[0] + z] = new THREE.Vector3(x, y, z).distanceTo(center);
+        }
+      }
+    }
+  }
+
+  const output = physics.marchingCubes(dims, potential, shift, scale);
+
+  let geometry = new THREE.BufferGeometry();
+  geometry.setIndex([...output.faces]);
```

Comment:
I didn't know you could do this. I thought `BufferGeometry.setIndex` takes `BufferAttribute`. That's what the docs say: https://threejs.org/docs/#api/en/core/BufferGeometry.setIndex

It looks like doing it this was creates another pointless copy. Why not just create the `BufferAttribute` directly? What is the reason?

Reply:
Index is different from other attributes. So an integer array can be used directly. But we can take a more canonical way